test/endpoint_testrunner.py

```diff
@@ -0,0 +1,238 @@
+        if inf_train["inf"]:
+            versions.append(version)
+
+    # Sort ascending to descending, use lambda to ensure 2.2 < 2.15, for instance
+    versions.sort(key=lambda version_str: [int(point) for point in version_str.split(".")], reverse=True)
+
+    framework_versions = versions if len(versions) < 4 else versions[:3]
+    image_definitions = []
+    for fw_version in framework_versions:
+        if fw_version in pre_populated_py_version:
+            py_versions = pre_populated_py_version[fw_version]
+        else:
+            py_versions = [get_canary_default_tag_py3_version(framework, fw_version)]
+        for py_version in py_versions:
+            image_definition = []
+            if processor == "gpu" or processor is None:
+                image_definitions.append({
+                            "registry": registry,
+                            "region": region,
+                            "repository": f"{framework}-{image_type}",
+                            "framework_version": fw_version,
+                            "processor": processor,
+                            "py_version": py_version
+                        })
+            if processor == "cpu" or processor is None:
+                image_definitions.append({
+                            "registry": registry,
+                            "region": region,
+                            "repository": f"{framework}-{image_type}",
+                            "framework_version": fw_version,
+                            "processor": processor,
+                            "py_version": py_version
+                        })
+
+    return image_definitions
+
+
+def execute_endpoint_test(framework_name, image_definition, account_id, sagemaker_region, registry, region):
+    
+    repository = image_definition.get("repository")
+    framework_version = image_definition.get("framework_version")
+    processor = image_definition.get("processor")
+    python_version = image_definition.get("py_version")
+    tag = f"{framework_version}-{processor}-{python_version}"
+
+    retries = "" # " --reruns 2"
+    instance_type = "p2.xlarge"
+    python_version = ("2" if "27" in python_version else
+                      "3" if "36" in python_version else
+                      python_version.lstrip("py"))
+
+    test_location = os.path.join("test", "sagemaker_endpoint_tests", framework_name)
+
+    ctx = Context()
+    with ctx.cd(test_location):
+        run_out = ctx.run(f"pytest -rs{retries} test_endpoint.py "
+            f"--account-id {account_id} "
+            f"--region {region} "
+            f"--registry {registry} "
+            f"--repository {repository} "
+            f"--instance-type ml.{instance_type} "
+            f"--framework-version {framework_version} "
+            f"--sagemaker-region {sagemaker_region} "
+            f"--py-version {python_version} "
+            f"--tag {tag}",
+            warn=True, echo=True
```

Comment:
why add warn=True here? And since we're not executing this via multiprocessing, is there any downside to using the `--junitxml` flag directly from pytest?